src/SelfServe/Decorators.tsx

```diff
@@ -8,37 +12,91 @@ interface Decorator {
 }
 
 interface InputOptionsBase {
+  /**
+   * Translation key corresponding to the label of the UI element
+   */
   labelTKey: string;
 }
 
+/**
+ * Numeric input UI element is rendered.
+ */
 export interface NumberInputOptions extends InputOptionsBase {
+  /**
+   * Min value of the numeric input UI element
+   */
   min: (() => Promise<number>) | number;
+  /**
+   * Max value of the numeric input UI element
+   */
   max: (() => Promise<number>) | number;
+  /**
+   * Value by which the numeric input is incremented or decremented in the UI.
+   */
   step: (() => Promise<number>) | number;
+  /**
+   * The type of the numeric input UI element
+   */
   uiType: NumberUiType;
 }
 
+/**
+ * Text box is rendered.
+ */
 export interface StringInputOptions extends InputOptionsBase {
+  /**
+   * Translation key corresponding to the place holder text of the text box.
+   */
   placeholderTKey?: (() => Promise<string>) | string;
 }
 
+/**
+ * Toggle is rendered.
+ */
 export interface BooleanInputOptions extends InputOptionsBase {
+  /**
+   * Translation key corresponding to the true label of the toggle
+   */
   trueLabelTKey: (() => Promise<string>) | string;
+  /**
+   * Translation key corresponding to the false label of the toggle
+   */
   falseLabelTKey: (() => Promise<string>) | string;
 }
 
+/**
+ * Dropdown is rendered.
+ */
 export interface ChoiceInputOptions extends InputOptionsBase {
+  /**
+   * Choices to be shown in the dropdown
+   */
   choices: (() => Promise<ChoiceItem[]>) | ChoiceItem[];
+  /**
+   * Translation key corresponding to the placeholder text of the dropdown.
+   */
   placeholderTKey?: (() => Promise<string>) | string;
 }
 
+/**
+ * Text is rendered.
+ */
 export interface DescriptionDisplayOptions {
+  /**
+   * Optional label for the text description
+   */
```

Comment:
Optional heading for the sentence/paragraph displayed by this description element.